src/main/java/randoop/reflection/WeightedConstantsOperationModel.java

```diff
@@ -0,0 +1,392 @@
+
+import java.lang.reflect.Constructor;
+import java.lang.reflect.Modifier;
+import java.util.*;
+import randoop.contract.CompareToAntiSymmetric;
+import randoop.contract.CompareToEquals;
+import randoop.contract.CompareToReflexive;
+import randoop.contract.CompareToSubs;
+import randoop.contract.CompareToTransitive;
+import randoop.contract.EqualsHashcode;
+import randoop.contract.EqualsReflexive;
+import randoop.contract.EqualsReturnsNormally;
+import randoop.contract.EqualsSymmetric;
+import randoop.contract.EqualsToNullRetFalse;
+import randoop.contract.EqualsTransitive;
+import randoop.contract.ObjectContract;
+import randoop.generation.ComponentManager;
+import randoop.main.ClassNameErrorHandler;
+import randoop.operation.*;
+import randoop.sequence.Sequence;
+import randoop.test.ContractSet;
+import randoop.types.ClassOrInterfaceType;
+import randoop.types.Type;
+import randoop.util.MultiMap;
+
+/**
+ * {@code WeightedConstantsOperationModel} represents the information context from which tests are
+ * generated. It is also used to store information needed for constant mining. The model includes:
+ *
+ * <ul>
+ *   <li>classes under test,
+ *   <li>operations of all classes,
+ *   <li>any atomic code sequences derived from command-line arguments, and
+ *   <li>the contracts or oracles used to generate tests.
+ * </ul>
+ *
+ * <p>This class manages all information about generic classes internally, and instantiates any type
+ * variables in operations before returning them.
+ */
+public class WeightedConstantsOperationModel extends AbstractOperationModel {
+
+  /** The set of class declaration types for this model */
+  private Set<ClassOrInterfaceType> classTypes;
+
+  /** The count of classes under test for this model */
+  private int classCount;
+
+  /** The set of input types for this model */
+  private Set<Type> inputTypes;
+
+  /** The set of class objects used in the exercised-class test filter */
+  private final LinkedHashSet<Class<?>> exercisedClasses;
+
+  /** Map for singleton sequences of literals extracted from classes. */
+  private MultiMap<ClassOrInterfaceType, Sequence> classLiteralMap;
+
+  /** The term frequency mapping for each sequence in the set of all classes */
+  private Map<Sequence, Integer> tfFrequency;
+
+  /** Set of singleton sequences for values from TestValue annotated fields. */
+  private Set<Sequence> annotatedTestValues;
+
+  /** Set of object contracts used to generate tests. */
+  private ContractSet contracts;
+
+  /** Set of concrete operations extracted from classes */
+  private Set<TypedOperation> operations;
+
+  /** Create an empty model of test context. */
+  private WeightedConstantsOperationModel() {
+    classTypes = new LinkedHashSet<>();
+    inputTypes = new LinkedHashSet<>();
+    classLiteralMap = new MultiMap<>();
+
+    annotatedTestValues = new LinkedHashSet<>();
+    contracts = new ContractSet();
+    contracts.add(EqualsReflexive.getInstance());
```

Comment:
We essentially did these copy-paste duplications + created `AbstractOperationModel.java` because we couldn't just extend 'OperationModel.java' without changing its factory constructor.